tasks/server-tests/vitest.setup.mts

```diff
@@ -0,0 +1,130 @@
+import { fileURLToPath } from 'node:url'
+
+import { afterAll, afterEach, beforeAll, expect } from 'vitest'
+import { fs, path, $ } from 'zx'
+import type { ProcessPromise } from 'zx'
+
+import { getConfig } from '@redwoodjs/project-config'
+
+$.verbose = !!process.env.VERBOSE
+
+type TestContext = {
+  p?: ProcessPromise
+  projectConfig: ReturnType<typeof getConfig>
+}
+export const testContext: TestContext = {
+  // Casting here because `beforeAll` below sets this and this file runs before all tests.
+  // Working around it being possibly undefined muddies the code in the tests.
+  projectConfig: {} as ReturnType<typeof getConfig>
+}
+
+const __dirname = fileURLToPath(new URL('./', import.meta.url))
+// @redwoodjs/cli (yarn rw)
+export const rw = path.resolve(__dirname, '../../packages/cli/dist/index.js')
+// @redwoodjs/api-server (yarn rw-server)
+export const rwServer = path.resolve(__dirname, '../../packages/api-server/dist/bin.js')
+// @redwoodjs/web-server (yarn rw-web-server)
+export const rwWebServer = path.resolve(__dirname, '../../packages/web-server/dist/bin.js')
+
+// When running `yarn vitest run` to run all the test suites, log the bin paths only once.
+if (!globalThis.loggedBinPaths) {
+  console.log([
```

Comment:
I'm getting this output when I run the tests locally. Is `undefined` expected?
![image](https://github.com/redwoodjs/redwood/assets/30793/93ec8cee-00a4-4c7f-86db-b4eaef94880c)

Reply:
No I realize I made a mistake by moving this out of the beforeAll, will move it back

Reply:
Fixed in https://github.com/redwoodjs/redwood/pull/9958/commits/5e7fc94f1a2cbfa0b6f16b9e9c29ba02469507f0